### PR TITLE
Feat: create grpc base for invite service

### PR DESCRIPTION
**Pull request by skelet0no**
1. Create grpc base
2. Add methods tp grpc
3. Add messages to proto file

### CODE REVIEW

File: invite_service/main.py [hunk 1]
```diff
@@ -1,2 +1,14 @@
...
+import grpc
+from concurrent import futures
+
+import proto.invite_service_pb2_grpc as invite_service_grpc
+from src.invite_service_impl import InviteServiceImpl
+
 if __name__ == "__main__":
-    print("PyCharm")
+    logging.basicConfig()
+    server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
```
Comment (Nhsdkk) [inline invite_service/main.py]: Switch to async server

File: invite_service/proto/invite_service.proto [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  optional string message = 3;
+}
+
+message BaseResponse {
+  int64 code = 1;
+  optional string message = 2;
+}
+
+message DeleteInviteRequest {
+  string id = 1;
```
Comment (Nhsdkk) [inline invite_service/proto/invite_service.proto]: Rename to invite_id

File: invite_service/proto/invite_service.proto [hunk 3]
```diff
@@ -0,0 +1,53 @@
+syntax = "proto3";
+
+service InviteService {
+  rpc get_invites_by_user_id(InvitesByUserIdRequest) returns (InvitesResponse){}
+  rpc create_invite(Invite) returns (BaseResponse){}
+  rpc update_invite(Invite) returns (BaseResponse){}
+  rpc delete_invite(DeleteInviteRequest) returns (BaseResponse){}
```
Comment (Nhsdkk) [inline invite_service/proto/invite_service.proto:8]: Missing get_invites_by_invitee_id

File: invite_service/src/invite_service_impl.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        InvitesResponse
+            Invites object for event response.
+        """
+        return proto.InvitesResponse(
+            code=200,
+            invites=proto.ListOfInvites(
+                invites=[invite.to_grpc_invite() for invite in self.invites]
+            ),
+        )
+
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py:69]: Missing get_invites_by_invitee_id

File: invite_service/src/models/invite.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+from dataclasses import dataclass
+from proto.invite_service_pb2 import Invite as GrpcInvite, InviteStatus
+
+
+@dataclass
+class Invite:
```
Comment (Nhsdkk) [inline invite_service/src/models/invite.py:7]: No docstring here for Invite class

File: invite_service/src/models/invite.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+from dataclasses import dataclass
```
Comment (Nhsdkk) [inline invite_service/src/models/invite.py:2]: No docstring for file

File: invite_service/src/models/invite.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+@dataclass
+class Invite:
+    id: str
+    event_id: str
+    author_id: str
+    invitee_id: str
+    status: InviteStatus
+
+    def to_grpc_invite(self) -> GrpcInvite:
```
Comment (Nhsdkk) [inline invite_service/src/models/invite.py:38]: No docstring for function

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Check review and make changes accordingly

File: invite_service/src/invite_service_impl.py [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+    Implementation of the Event Service.
+
+    Attributes
+    ----------
+    invites : List[Invite]
+        List of invites
+
+    Methods
+    -------
+    get_invites_by_user_id()
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Add function args

File: invite_service/src/invite_service_impl.py [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+        Function that need to be bind to the server that returns invites list.
+    create_invite()
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Add function args

File: invite_service/src/invite_service_impl.py [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+        Function that need to be bind to the server that creates the invite.
+    update_invite()
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Add function args

File: invite_service/src/invite_service_impl.py [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+        Function that need to be bind to the server that updates the invite.
+    delete_invite()
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Add function args

File: invite_service/src/invite_service_impl.py [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+        Function that need to be bind to the server that deletes the invite.
+    """
+
+    def __init__(self):
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Add return type

File: invite_service/src/invite_service_impl.py [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+    -------
+    get_invites_by_user_id()
+        Function that need to be bind to the server that returns invites list.
+    create_invite()
+        Function that need to be bind to the server that creates the invite.
+    update_invite()
+        Function that need to be bind to the server that updates the invite.
+    delete_invite()
+        Function that need to be bind to the server that deletes the invite.
+    """
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:32]: Add empty string at the end of docstring

File: invite_service/src/invite_service_impl.py [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+
+    def get_invites_by_user_id(
+        self, request: proto.InvitesByUserIdRequest, context: grpc.ServicerContext
+    ) -> proto.InvitesResponse:
+        """
+        Get all invites by user id.
+
+        Parameters
+        ----------
+        request : InvitesByUserIdRequest
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Args types in annotations and docstring are different. Add "proto." in docstring to fix

File: invite_service/src/invite_service_impl.py [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+            Request data.
+        context : grpc.ServicerContext
+            Request context.
+
+        Returns
+        -------
+        InvitesResponse
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Args types in annotations and docstring are different. Add "proto." in docstring to fix

File: invite_service/src/invite_service_impl.py [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+            Invites object for invite response.
+        """
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:62]: Add empty line at the end of docstring

File: invite_service/src/invite_service_impl.py [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+
+    def get_invites_by_invitee_id(
+        self, request: proto.GetInviteeByInviteIdRequest, context: grpc.ServicerContext
+    ) -> proto.InvitesResponse:
+        """
+        Get all invites by invitee id.
+
+        Parameters
+        ----------
+        request : InviteeByInviteIdRequest
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Args types in annotations and docstring are different. Add "proto.Get" in docstring to fix

File: invite_service/src/invite_service_impl.py [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+            Request data.
+        context : grpc.ServicerContext
+            Request context.
+
+        Returns
+        -------
+        InvitesResponse
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Args types in annotations and docstring are different. Add "proto." in docstring to fix

File: invite_service/src/invite_service_impl.py [hunk 19]
```diff
@@ -0,0 +1,144 @@
...
+            Invites object for invite response.
+        """
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:88]: Add empty string at the end of docstring

File: invite_service/src/invite_service_impl.py [hunk 20]
```diff
@@ -0,0 +1,144 @@
...
+
+    def create_invite(
+        self, request: proto.Invite, context: grpc.ServicerContext
+    ) -> proto.BaseResponse:
+        """
+        Create invite.
+
+        Parameters
+        ----------
+        request : GrpcInvite
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Args types in annotations and docstring are different. Remove "GrpcInvite", add "proto.Invite" instead in docstring to fix

File: invite_service/src/invite_service_impl.py [hunk 21]
```diff
@@ -0,0 +1,144 @@
...
+            Request data containing GrpcInvite.
+        context : grpc.ServicerContext
+            Request context.
+
+        Returns
+        -------
+        BaseResponse
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Args types in annotations and docstring are different. Add "proto." in docstring to fix

File: invite_service/src/invite_service_impl.py [hunk 22]
```diff
@@ -0,0 +1,144 @@
...
+            Object containing status code and message if the response status is not 200.
+        """
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:109]: Add empty line at the end of docstring

File: invite_service/src/invite_service_impl.py [hunk 23]
```diff
@@ -0,0 +1,144 @@
...
+
+    def delete_invite(
+        self, request: proto.DeleteInviteRequest, context: grpc.ServicerContext
+    ) -> proto.BaseResponse:
+        """
+        Delete invite.
+
+        Parameters
+        ----------
+        request : DeleteInviteRequest
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Args types in annotations and docstring are different. Add "proto." in docstring to fix

File: invite_service/src/invite_service_impl.py [hunk 24]
```diff
@@ -0,0 +1,144 @@
...
+            Request data containing invite ID.
+        context : grpc.ServicerContext
+            Request context.
+
+        Returns
+        -------
+        BaseResponse
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Args types in annotations and docstring are different. Add "proto." in docstring to fix

File: invite_service/src/invite_service_impl.py [hunk 25]
```diff
@@ -0,0 +1,144 @@
...
+            Object containing status code and message if the response status is not 200.
+        """
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:151]: Add empty string at the end of docstring

File: invite_service/src/invite_service_impl.py [hunk 26]
```diff
@@ -0,0 +1,144 @@
...
+        Parameters
+        ----------
+        request : GrpcInvite
+            Request data containing GrpcInvite.
+        context : grpc.ServicerContext
+            Request context.
+
+        Returns
+        -------
+        BaseResponse
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Args types in annotations and docstring are different. Add "proto." in docstring to fix

File: invite_service/src/invite_service_impl.py [hunk 27]
```diff
@@ -0,0 +1,144 @@
...
+            Object containing status code and message if the response status is not 200.
+        """
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:130]: Add empty line at the end of docstring

File: invite_service/src/invite_service_impl.py [hunk 28]
```diff
@@ -0,0 +1,144 @@
...
+
+    def update_invite(
+        self, request: proto.Invite, context: grpc.ServicerContext
+    ) -> proto.BaseResponse:
+        """
+        Update invite.
+
+        Parameters
+        ----------
+        request : GrpcInvite
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Args types in annotations and docstring are different. Remove "GrpcInvite", add "proto.Invite" instead in docstring to fix

File: invite_service/src/models/invite.py [hunk 29]
```diff
@@ -0,0 +1,55 @@
...
+        ID of the author
+    invitee_id : str
+        ID of the invitee
+    status: InviteStatus:
+        Invite status
+
+
+    Methods
+    -------
+    to_grpc_invite()
```
Comment (ZixMai) [inline invite_service/src/models/invite.py:27]: Add arguments to method in docstring

File: invite_service/src/models/invite.py [hunk 30]
```diff
@@ -0,0 +1,55 @@
...
+
+    def to_grpc_invite(self) -> GrpcInvite:
+        """
+        Converts invite information to GrpcInvite
+
+        Returns
+        -------
+        GrpcInvite
+            Invite data in GrpcInvite instance
+        """
```
Comment (ZixMai) [inline invite_service/src/models/invite.py:47]: Add empty line at the end of docstring

**Review by ZixMai — CHANGES_REQUESTED** [review]
Big changes in docstrings needed ...

File: invite_service/main.py [hunk 31]
```diff
@@ -1,2 +1,25 @@
...
+import asyncio
+import sys
+
+import grpc
+
+import proto.invite_service_pb2_grpc as invite_service_grpc
+from src.invite_service_impl import InviteServiceImpl
+
+
+async def serve():
```
Comment (ZixMai) [inline invite_service/main.py]: Add return type

**Review by ZixMai — CHANGES_REQUESTED** [review]

**Review by ZixMai — APPROVED** [review]

**Review by Nhsdkk — APPROVED** [review]
LGTM

File: invite_service/main.py (hunk 31 above)
Comment (Nhsdkk) [reply invite_service/main.py]: Should return None

**Review by MaxPlays35 — APPROVED** [review]

File: invite_service/src/models/invite.py [hunk 32]
```diff
@@ -0,0 +1,56 @@
...
+
+@dataclass
+class Invite:
+    """
+    Data class that stores user information
+
+    Attributes
+    ----------
+    id : str
+        ID of the user
```
Comment (Nhsdkk) [inline invite_service/src/models/invite.py]: Should be ID of the event

File: invite_service/src/models/invite.py [hunk 33]
```diff
@@ -0,0 +1,56 @@
+"""Invite Model"""
+from dataclasses import dataclass
+from proto.invite_service_pb2 import Invite as GrpcInvite, InviteStatus
+
+
+@dataclass
+class Invite:
+    """
+    Data class that stores user information
```
Comment (Nhsdkk) [inline invite_service/src/models/invite.py]: Should be "Data class that stores invite information"

File: invite_service/src/models/invite.py [hunk 34]
```diff
@@ -0,0 +1,56 @@
...
+        """
+        Converts invite information to GrpcInvite
+
+        Returns
+        -------
+        GrpcInvite
+            Invite data in GrpcInvite instance
+
+        """
+        invite = GrpcInvite(
```
Comment (Nhsdkk) [inline invite_service/src/models/invite.py]: Make return statement here

File: invite_service/src/invite_service_impl.py [hunk 35]
```diff
@@ -0,0 +1,150 @@
...
+    -------
+    get_invites_by_user_id(request, context)
+        Function that need to be bind to the server that returns invites list.
+    create_invite(request, context)
+        Function that need to be bind to the server that creates the invite.
+    update_invite(request, context)
+        Function that need to be bind to the server that updates the invite.
+    delete_invite(request, context)
+        Function that need to be bind to the server that deletes the invite.
+
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py:31]: Where is docs for get_invites_by_invitee_id

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Check my review

File: invite_service/src/models/invite.py (hunk 32 above)
Comment (Nhsdkk) [reply invite_service/src/models/invite.py]: invite sorry

**Review by Nhsdkk — APPROVED** [review]
LGTM

**Pull request merged**